src/Avalonia.Controls/SizeChangedEventArgs.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Avalonia.Interactivity;
+using Avalonia.Layout;
+using Avalonia.Utilities;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Provides data specific to a SizeChanged event.
+    /// </summary>
+    public class SizeChangedEventArgs : RoutedEventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent)
```

Comment:
I wonder if these events should have internal ctor. Like it was discussed here https://github.com/AvaloniaUI/Avalonia/pull/8860
But either way it's not important right now

Reply:
That conversation doesn't make sense to me. There is little harm in having these public -- this API isn't going to change. More though, how are apps supposed to implement this event themselves if constructors are private? I never understood why you are hiding constructors.